src/entropy.rs

```diff
@@ -0,0 +1,407 @@
+//! Summary statistics (e.g. mean, variance, etc.).
+use ndarray::{Array, ArrayBase, Data, Dimension, Zip};
+use num_traits::Float;
+use crate::errors::ShapeMismatch;
+
+/// Extension trait for `ArrayBase` providing methods
+/// to compute information theory quantities
+/// (e.g. entropy, Kullback–Leibler divergence, etc.).
+pub trait EntropyExt<A, S, D>
+    where
+        S: Data<Elem = A>,
+        D: Dimension,
+{
+    /// Computes the [entropy] *S* of the array values, defined as
+    ///
+    /// ```text
+    ///       n
+    /// S = - ∑ xᵢ ln(xᵢ)
+    ///      i=1
+    /// ```
+    ///
+    /// If the array is empty, `None` is returned.
+    ///
+    /// **Panics** if any element in the array is negative.
```

Comment:
```suggestion
    /// **Panics** if `ln` of any element in the array panics (which can occur for negative values for some `A`).
```

Just a clarification that it won't always panic for negative numbers. (`ln` of negative `f32`/`f64` is `NaN`.)